average_points[] points_array